src/shared/dm-util.c

```diff
@@ -43,3 +43,39 @@ int dm_deferred_remove(const char *name) {
         return 0;
 #endif
 }
+
+int dm_deferred_remove_cancel(const char *name) {
+        _cleanup_close_ int fd = -1;
+        _cleanup_free_ struct dm_ioctl *dm = NULL;
+        const uint32_t dm_size = sizeof(struct dm_ioctl) + sizeof(struct dm_target_msg) + strlen("@cancel_deferred_remove") + 1;
```

Comment:
use `size_t` as type here, since this is a size in memory you calculate here

There's no benefit in the `const` here, it's a local variable anyway. `const` is great to define contracts with outside callers or if it affects how data is laid out in the ELF file, but here it has no such effect, and it's hence just decoration...

Also, please use `STRLEN()` here, so that this actually *is* a constant expression

Reply:
Sure thing - used uint32_t as that's what dm_ioctl declares it as